plugins/qeta-backend/src/service/upload/azureBlobStorage.ts

```diff
@@ -0,0 +1,73 @@
+import * as fs from 'fs';
+import { v4 as uuidv4 } from 'uuid';
+import { Config } from '@backstage/config';
+import { QetaStore } from '../../database/QetaStore';
+import { Attachment } from '@drodil/backstage-plugin-qeta-common';
+import { File } from '../types';
+import { getAzureBlobServiceClient } from '../util';
+import { AttachmentStorageEngine, AttachmentStorageEngineOptions } from './attachmentStorageEngine';
+
+class AzureBlobStorageEngine implements AttachmentStorageEngine {
+  config: Config;
+  database: QetaStore;
+  backendBaseUrl: string;
+  qetaUrl: string;
+  container: string;
+
+  constructor(opts: AttachmentStorageEngineOptions) {
+    this.config = opts.config;
+    this.database = opts.database;
+    this.backendBaseUrl = this.config.getString('backend.baseUrl');
+    this.qetaUrl = `${this.backendBaseUrl}/api/qeta/attachments`;
+    this.container =
+      this.config.getOptionalString('qeta.storage.blobStorageContainer') ||
+      'backstage-qeta-images';
+  }
+
+  handleFile = async (
+    file: File,
+    options?: { postId?: number; answerId?: number; collectionId?: number },
+  ): Promise<Attachment> => {
+    const imageUuid = uuidv4();
+    const filename = `image-${imageUuid}-${Date.now()}.${file.ext}`;
+
+    const imageURI = `${this.qetaUrl}/${imageUuid}`;
+    const client = getAzureBlobServiceClient(this.config);
+    const container = client.getContainerClient(this.container);
+    if (!(await container.exists())) {
+        await container.create();
+    }
+
+    await container.uploadBlockBlob(filename, fs.createReadStream(file.path), file.size);
+
+    return await this.database.postAttachment({
+      uuid: imageUuid,
+      locationType: 'azure',
+      locationUri: imageURI,
+      extension: file.ext,
+      mimeType: file.mimeType,
+      path: filename,
+      binaryImage: Buffer.from('0'),
+      creator: '', // required to run locally on sqlite, otherwise it complains about null.
```

Comment:
Hm, does it really need to be empty string? How is it handled in other engines?

Reply:
Hm, weird it's empty string in other engines as well 🤔

Reply:
The field seems to suggest that it should be a reference to the user that uploaded the attachment. I guess I could put in an entity ref to the user entity?

Reply:
Yes, it should be the user reference

Reply:
Looks like I don't have this information available at this point. Perhaps this ties into the other issue with regards to using `Attachment` instead of `File`?

Reply:
Username should be fetched in the `attachments.ts` route handler like here https://github.com/drodil/backstage-plugin-qeta/blob/main/plugins/qeta-backend/src/service/routes/posts.ts#L45 and then passed to the engine. I think this has been missing for quite long now :D

Reply:
Fixed that now :)